### PR TITLE
Feature/2827 blui inline

### LINKED ISSUE #31: Make buttons with inline icons style properly
Buttons with icons in them have always been a huge problem to get the styling correct. So why not bring it to our themes?

Maybe for `<mat-icon>` with an `inline` attribute:

```html
<button mat-stroked-button color="primary">
    <mat-icon inline>favorite</mat-icon>
    Inline Icon
</button>
```

instead of styling it like 

image

They should look like [what Material Design suggests](https://material.io/components/buttons#specs)

image

- Ignore the `min-width` part, just focus on the margins
- also make icon size 20x20 to match React

----
### Additional context
- This would be a breaking change
- Do not touch `<mat-icon>` without the `inline` attribute — they are used for regular icon buttons and you wouldn't want to mess up the style

### PR DESCRIPTION
**Pull request by emclaug2**
Fixes #31 

#### Changes proposed in this Pull Request:
- Add `blui-inline` styles

```
<button mat-flat-button blui-inline color="accent">
  <mat-icon >account_circle</mat-icon>
  W/ Inline
</button>
```

#### Screenshots:
image

### CODE REVIEW

**Comment by emclaug2** [comment]
I'll make a feature request in the angular material rep to see if they'd consider adding something like this to their button.

**Comment by emclaug2** [comment]
Ok put a feature request up here: https://github.com/angular/components/issues/24280

File: _common.scss [hunk 1]
```diff
@@ -56,4 +56,21 @@
...
+    .mat-button-base[blui-inline] {
+        .mat-button-wrapper {
+            display: flex;
+            align-items: center;
+        }
+        .mat-icon {
+            height: 18px;
+            width: 18px;
+            font-size: 18px;
+            margin: 0 8px 0 -4px;
```
Comment (huayunh) [inline _common.scss:70]: Vertical distance should be in `rem` units

**Review by huayunh — COMMENTED** [review]
I didn't run the code locally but this is already looking awesome from the screenshot. 

_Shedding tears after seeing so many crooked buttons_

**Comment by huayunh** [comment]
Per the issue's description — React's default is 20x20. Material Design's default is 18x18. If you force me to pick among the two, I like 18x18 slightly better, but I don't have a strong opinion one way or the other. If we stick with 18x18, would you make an enhancement on the React theme side?

**Comment by emclaug2** [comment]
> Per the issue's description — React's default is 20x20. Material Design's default is 18x18. If you force me to pick among the two, I like 18x18 slightly better, but I don't have a strong opinion one way or the other. If we stick with 18x18, would you make an enhancement on the React theme side?

You got it! 
I prefer the 18x18 too, I've made the issue:
https://github.com/brightlayer-ui/react-themes/issues/24

File: _common.scss (hunk 1 above)
Comment (emclaug2) [reply _common.scss:70]: Hmm, for this I don't think it makes to make height in rem but width in px.  We'd want the icon to stay as a square. Should they both be in rem?

Comment (huayunh) [reply _common.scss:70]: Ah yes... that's what I meant to say. 🙈

**Review by andrupu — APPROVED** [review]

File: _common.scss (hunk 1 above)
Comment (emclaug2) [reply _common.scss]: This is now this is updated. I'll merge this in.

**Pull request merged**